sdk/samples/router/general.py

```diff
@@ -0,0 +1,265 @@
+    resource_request = ResourceRequest(
+        min_replica=0,
+        max_replica=2,
+        cpu_request="500m",
+        memory_request="512Mi"
+    )
+
+    # # Create a log config for the router
+    log_config = LogConfig(
+        result_logger_type=ResultLoggerType.NOP
+    )
+
+    # # Create an enricher for the router
+    enricher = Enricher(
+        image="asia.gcr.io/gods-dev/echo:1.0.2",
+        resource_request=ResourceRequest(
+            min_replica=0,
+            max_replica=2,
+            cpu_request="500m",
+            memory_request="512Mi"
+        ),
+        endpoint="/",
+        timeout="60ms",
+        port=8080,
+        env=[
+            EnvVar(
+                name="humans",
+                value="farmer-joe"
+            )
+        ]
+    )
+
+    # # Create an ensembler for the router
+    ensembler = DockerRouterEnsemblerConfig(
+        id=1,
+        image="asia.gcr.io/gods-dev/echo:1.0.2",
+        resource_request=ResourceRequest(
+            min_replica=1,
+            max_replica=3,
+            cpu_request="500m",
+            memory_request="512Mi"
+        ),
+        endpoint="/echo",
+        timeout="60ms",
+        port=8080,
+        env=[],
+    )
+
+    # # Create the RouterConfig instance
+    router_config = RouterConfig(
+        environment_name="id-dev",
+        name="what-does-the-fox-say",
+        routes=routes,
+        rules=rules,
+        default_route_id="test",
+        experiment_engine=experiment_config,
+        resource_request=resource_request,
+        timeout="100ms",
+        log_config=log_config,
+        enricher=enricher,
+        ensembler=ensembler
+    )
+
+    # 1. Create a new router using the RouterConfig object
+    new_router = turing.Router.create(router_config)
+
+    # 2. List all routers
+    routers = turing.Router.list()
+    for r in routers:
+        if r.name == new_router.name:
+            my_router_id = r.id
+        print(r)
+
+    # 3. Get the router you just created using the router_id obtained
+    my_router = turing.Router.get(my_router_id)
+
+    # Access the router config from the returned Router object directly
+    my_router_config = my_router.config
+
+    # Modify something in the router config
+    my_router_config.routes.append(
+        Route(
+            id='fee-fi-fo-fum',
+            endpoint='http://fox-says.fee-fi-fo-fum',
+            timeout='20ms'
+        )
+    )
+
+    # 4. Update the router with the new router config
+    my_router.update(my_router_config)
+
+    # 5. List all the router config versions of your router
+    my_router_versions = my_router.list_versions()
+    for ver in my_router_versions:
+        if ver.status == 'undeployed':
+            first_ver_no = ver.version
+        if ver.status == "deployed":
+            latest_ver_no = ver.version
+        print(ver)
+
+    # 6. Deploy a specific router config version (the first one we created)
```

Comment:
I would suggest for points 6-8 that you print some debugging information to feedback to the user what is happening when you undeploy and deploy.